### ISSUE TITLE
XML parse error for the base facet  has a greater min length ...

### ISSUE THREAD

**Issue by vivek3011** [sissaschool/xmlschema#396]
![Screenshot_2024-03-21-18-10-49-19_6012fa4d4ddec268fc5c7112cbb265e7](https://github.com/sissaschool/xmlschema/assets/65756475/98086abf-17c9-4f7d-b357-dba4f5c1fcf7)
This is the part of my code :

```xsd
<element name="add_list_of _strings" minOccurs="0"> 
                              <simpleType>
                                 <restriction>
                                    <simpleType>
                                       <list>
                                          <simpleType>
                                             <restriction base="string">
                                             <minLength value="5"/>
                                             <maxLength value="6"/>
                                          </restriction>
                                          </simpleType>
                                       </list>
                                    </simpleType>
                                    <minLength value="1"/>
                                    <maxLength value="6"/>
                                 </restriction>
                              </simpleType>
                           </element>
```

The error:
XMLSchemaParseError: base facet has a greater min length (5):

Schema component:

  <minLength xmlns="http://www.w3.org/2001/XMLSchema" value="1" />

I am not able to parse the this xsd file because I got an error over this element . I don't where the error is it in element or in  XMLschema library..... I want to where it went wrong and if it is wrong in the element let me what it is I should I correct this  and reason for the error and with the valid proof  
Please help me with this error and give me detail information about the error

**Comment by vivek3011** [comment]
![Screenshot_2024-03-21-18-10-49-19_6012fa4d4ddec268fc5c7112cbb265e7](https://github.com/sissaschool/xmlschema/assets/65756475/6e9098ab-1e90-427a-b6cf-170da2c93563)

**Comment by brunato** [comment]
Hi,

I have fixed this bug in minor release v3.2. 

The fix has required a small refactoring of the class used for representing `<xs:list>` definitions (XsdList). It wasn't the only solution for this, but it's the simplest and it didn't break any test, including W3C tests. 

thank you for reporting the problem
